packages/core/src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -179,9 +182,13 @@ export default class GooglePayPaymentProcessor {
                             apiVersion: paymentDataRequest.apiVersion,
                             apiVersionMinor: paymentDataRequest.apiVersionMinor,
                         })
-                        .then((response) => {
-                            if (response.result) {
-                                return;
+                        .then(({ result: isReadyToPay }) => {
+                            if (isReadyToPay) {
+                                return isReadyToPay;
+                            }
+
+                            if (this._isInAppUnsupportedBrowser('instagram')) {
```

Comment:
Does problem occurs only with `instagram` user agent? I think would be better to do it for all in-app browsers

Reply:
@vitalii-koshovyi I didn't check it for other in-app browsers. But i will check what happens if we use facebook for example and if we have the same issue i will make some changes and let you know. Thanks for the point!